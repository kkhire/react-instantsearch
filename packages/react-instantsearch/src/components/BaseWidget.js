import React from 'react';
import PropTypes from 'prop-types';

const BaseWidget = ({
  cx,
  children,
  canRefine,
  renderHeader,
  renderFooter,
}) => (
  <div className={cx('', !canRefine && '-noRefinement')}>
    {renderHeader && (
      <div className={`${cx('header')} ais-header`}>{renderHeader()}</div>
    )}

    <div className={`${cx('body')} ais-body`}>{children}</div>

    {renderFooter && (
      <div className={`${cx('footer')} ais-footer`}>{renderFooter()}</div>
    )}
  </div>
);

BaseWidget.propTypes = {
  cx: PropTypes.func.isRequired,
  children: PropTypes.node.isRequired,
  canRefine: PropTypes.bool.isRequired,
  renderHeader: PropTypes.func,
  renderFooter: PropTypes.func,
};

export default BaseWidget;
