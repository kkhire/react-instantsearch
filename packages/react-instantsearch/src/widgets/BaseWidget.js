import React from 'react';
import PropTypes from 'prop-types';

const BaseWidget = ({
  cx,
  children,
  cantRefine,
  renderHeader,
  renderFooter,
}) => (
  <div className={cx('', cantRefine && `-noRefinement`)}>
    {renderHeader && (
      <div className={`${cx('header')} ais-header`}>{renderHeader()}</div>
    )}

    <div className={`${cx('body')} ais-body`}>{children}</div>

    {renderFooter && (
      <div className={`${cx('footer')} ais-footer`}>{renderFooter()}</div>
    )}
  </div>
);

BaseWidget.propTypes = {
  cx: PropTypes.func.isRequired,
  children: PropTypes.node.isRequired,
  cantRefine: PropTypes.bool,
  renderHeader: PropTypes.func,
  renderFooter: PropTypes.func,
};

export default BaseWidget;
