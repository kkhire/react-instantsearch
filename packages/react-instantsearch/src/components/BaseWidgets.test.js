import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import BaseWidget from './BaseWidget';

Enzyme.configure({ adapter: new Adapter() });

describe('BaseWidget', () => {
  const defaultProps = {
    cx: (...args) => args.filter(x => Boolean(x)).join(' '),
    canRefine: true,
  };

  it('expect to render', () => {
    const props = {
      ...defaultProps,
    };

    const wrapper = shallow(
      <BaseWidget {...props}>
        <div>Hello content</div>
      </BaseWidget>
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('expect to render without refinement', () => {
    const props = {
      ...defaultProps,
      canRefine: false,
    };

    const wrapper = shallow(
      <BaseWidget {...props}>
        <div>Hello content</div>
      </BaseWidget>
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('expect to render with header', () => {
    const props = {
      ...defaultProps,
      renderHeader: () => 'Header',
    };

    const wrapper = shallow(
      <BaseWidget {...props}>
        <div>Hello content</div>
      </BaseWidget>
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('expect to render with footer', () => {
    const props = {
      ...defaultProps,
      renderFooter: () => 'Footer',
    };

    const wrapper = shallow(
      <BaseWidget {...props}>
        <div>Hello content</div>
      </BaseWidget>
    );

    expect(wrapper).toMatchSnapshot();
  });
});
