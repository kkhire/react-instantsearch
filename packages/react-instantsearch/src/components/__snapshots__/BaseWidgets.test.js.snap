// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseWidget expect to render 1`] = `
<div
  className=""
>
  <div
    className="body ais-body"
  >
    <div>
      Hello content
    </div>
  </div>
</div>
`;

exports[`BaseWidget expect to render with footer 1`] = `
<div
  className=""
>
  <div
    className="body ais-body"
  >
    <div>
      Hello content
    </div>
  </div>
  <div
    className="footer ais-footer"
  >
    Footer
  </div>
</div>
`;

exports[`BaseWidget expect to render with header 1`] = `
<div
  className=""
>
  <div
    className="header ais-header"
  >
    Header
  </div>
  <div
    className="body ais-body"
  >
    <div>
      Hello content
    </div>
  </div>
</div>
`;

exports[`BaseWidget expect to render without refinement 1`] = `
<div
  className="-noRefinement"
>
  <div
    className="body ais-body"
  >
    <div>
      Hello content
    </div>
  </div>
</div>
`;
